src/net_processing.cpp

```diff
@@ -5739,7 +5750,51 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         }
                         if (tx_relay->m_bloom_filter && !tx_relay->m_bloom_filter->IsRelevantAndUpdate(*txinfo.tx)) continue;
                         // Send
-                        vInv.push_back(inv);
+                        bool fanout = true;
+                        const auto wtxid = txinfo.tx->GetWitnessHash();
+                        if (reconciles_txs) {
+                            auto txiter = m_mempool.GetIter(txinfo.tx->GetHash());
+                            if (txiter) {
+                                if ((*txiter)->GetCountWithDescendants() > 1) {
+                                    // If a transaction has in-mempool children, always fanout it.
+                                    // Until package relay is implemented, this is needed to avoid
+                                    // breaking parent+child relay expectations in some cases.
+                                    //
+                                    // Potentially reconciling parent+child would mean that for every
+                                    // child we need to to check if any of the parents is currently
+                                    // reconciled so that the child isn't fanouted ahead. But then
+                                    // it gets tricky when reconciliation sets are full: a) the child
+                                    // can't just be added; b) removing parents from reconciliation
+                                    // sets for this one child is not good either.
```

Comment:
I think the issue can be even worst if you're considering the receiver-initiated approach, where the receiver might have already the child from another peer and the BIP331's `getskgtxns` is in-flight. Both the parent and child wtxid reconciliation set entries are going to be wasted, I think.

I believe the best approach to combine it with package relay support for now is just to look if it doesn't raise privacy or CPU robustness approach for now (at least until one is finalized and deployed), and swallow the bullet in term of bandwidth savings performance.

Reply:
I assume what you're saying is that "your code works, but the problem it solves is harder than what the comment says". Correct me if I'm wrong.